### PR TITLE
bug/FP 1974 and 1976 No Whitespace Input on Excp/Ext Forms

### PR DESCRIPTION
**Pull request by sophia-massie**
## Overview

To audit form fields to make sure they're not only whitespace submissions

## Related

- [FP-1974](https://jira.tacc.utexas.edu/browse/FP-1974)
 - [FP-1976](https://jira.tacc.utexas.edu/browse/FP-1976)

## Changes

- Used Garrett's script in the other-exception, threshold-exception and extension form template
- Used script on textareas as well so that if more textarea fields are added in the future (besides justification), it will check those as well

## Testing

1. In Exception/views.py add the following snippet to the line 105 and 37

<details>

submitters = [ (2, 'TESTGOLD', 10000001, 'gmunoz1', 'CHCD'), (3, 'TESTGOLD', 10000002, 'gmunoz1', 'CHCD'), (1, 'TESTGOLD', 10000000, 'gmunoz1', 'CHCD') ]
</details>

2. Go to  [http://localhost:8000/submissions/threshold-exception/](http://localhost:8000/submissions/threshold-exception/) and [http://localhost:8000/submissions/other-exception/](http://localhost:8000/submissions/other-exception/)
3. Make sure page looks submits correctly and does not allow whitespace only as input. 
4. In extension/views.py add the following snippet to the line 24

<details>

submitters = [ (2, 'TESTGOLD', 10000001, 'gmunoz1', 'CHCD'), (3, 'TESTGOLD', 10000002, 'gmunoz1', 'CHCD'), (1, 'TESTGOLD', 10000000, 'gmunoz1', 'CHCD') ]
</details>

5. Go to  [http://localhost:8000/submissions/extension-request/](http://localhost:8000/submissions/extension-request/) 
6. Make sure page looks submits correctly and does not allow whitespace only as input. 

## UI

…

### CODE REVIEW

**Review by jmartinez-tacc — APPROVED** [review]
LGTM!

File: apcd-cms/src/apps/exception/templates/exception_submission_form/exception_other_form.html [hunk 1]
```diff
@@ -141,6 +141,16 @@ <h4>Acknowledgment of Terms</h4>
...
       {# Scripts #}
       <aside>
+        <!--To check for blank space only in text inputs #}-->
+        <script src="{% static 'utils/js/checkForBlankInputs.js' %}"></script>
+        {# Adds blank space listener to all text input fields on page load #}
+        <script>
+          const formTextInputs = document.querySelectorAll('input[type=text');
+          const formTextAreaInputs = document.querySelectorAll('textarea');
+          noEmptyInputs(formTextInputs);
+          noEmptyInputs(formTextAreaInputs);
```
Comment (edmondsgarrett) [inline apcd-cms/src/apps/exception/templates/exception_submission_form/exception_other_form.html:151]:
```suggestion
          const formTextInputs = document.querySelectorAll('input[type=text], textarea');
          noEmptyInputs(formTextInputs);
```

File: apcd-cms/src/apps/exception/templates/exception_submission_form/exception_threshold_form.html [hunk 2]
```diff
@@ -210,6 +210,15 @@ <h4>Acknowledgment of Terms</h4>
...
 
         <aside>
+          <!-- To check for blank space only in text inputs-->
+          <script src="{% static 'utils/js/checkForBlankInputs.js' %}"></script>
+          <!--Adds blank space listener to all text input fields on page load #-->
+          <script>
+          const formTextInputs = document.querySelectorAll('input[type=text');
+          const formTextAreaInputs = document.querySelectorAll('textarea');
+          noEmptyInputs(formTextInputs);
+          noEmptyInputs(formTextAreaInputs);
```
Comment (edmondsgarrett) [inline apcd-cms/src/apps/exception/templates/exception_submission_form/exception_threshold_form.html:220]:
```suggestion
          const formTextInputs = document.querySelectorAll('input[type=text], textarea');
          noEmptyInputs(formTextInputs);
```

File: apcd-cms/src/apps/extension/templates/extension_submission_form/extension_submission_form.html [hunk 3]
```diff
@@ -200,8 +200,16 @@ <h4>Acknowledgment of Terms</h4>
...
-      {# WARNING: Many are cruft that created this now-static markup, but not all #}
       <aside>
+        <!-- To check for blank space only in text inputs-->
+        <script src="{% static 'utils/js/checkForBlankInputs.js' %}"></script>
+        <!--Adds blank space listener to all text input fields on page load #-->
+        <script>
+          const formTextInputs = document.querySelectorAll('input[type=text');
+          const formTextAreaInputs = document.querySelectorAll('textarea');
+          noEmptyInputs(formTextInputs);
+          noEmptyInputs(formTextAreaInputs);
```
Comment (edmondsgarrett) [inline apcd-cms/src/apps/extension/templates/extension_submission_form/extension_submission_form.html:211]:
```suggestion
          const formTextInputs = document.querySelectorAll('input[type=text], textarea');
          noEmptyInputs(formTextInputs);
```

**Review by edmondsgarrett — CHANGES_REQUESTED** [review]
LGTM, working locally! I've included some cleanup suggestions.

File: apcd-cms/src/apps/extension/templates/extension_submission_form/extension_submission_form.html (hunk 3 above)
Comment (sophia-massie) [reply apcd-cms/src/apps/extension/templates/extension_submission_form/extension_submission_form.html]: Good catch, thank you!

**Review by edmondsgarrett — APPROVED** [review]
LGTM!

File: apcd-cms/src/apps/exception/templates/exception_submission_form/exception_other_form.html (hunk 1 above)
Comment (fnets) [reply apcd-cms/src/apps/exception/templates/exception_submission_form/exception_other_form.html]: nice catch!

**Review by fnets — APPROVED** [review]
LGTM! Thanks for the quick work

**Pull request merged**